File: Sr-Navalha-Back/src/main/java/com/fabricasoftware/SrNavalha/models/UsuarioCliente.java
package com.fabricasoftware.SrNavalha.models;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;
import java.time.LocalDate;


@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "usuario_cliente")
@ToString
public class UsuarioCliente {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nome;
    private String telefone;
    private String email;
    private LocalDate dataNascimento;
    private String cpf;
    private String tipo;
    @ManyToOne(cascade=CascadeType.PERSIST)
    private Endereco endereco;
    @OneToOne(cascade=CascadeType.PERSIST)
    private Credencial credencial;
}

File: Sr-Navalha-Front/src/app/models/usuario-cliente.ts
import { Credencial } from "./credencial";
import { Endereco } from "./endereco";

export class UsuarioCliente {
    id: any;
    nome: any;
    telefone: any;
    email: any;
    dataNascimento: any;
    cpf: any;
    tipo: any;
    endereco: Endereco = new Endereco;
    credencial: Credencial = new Credencial;
}